controllers/quadpid/quadpid.c

```diff
@@ -142,26 +148,13 @@ int ctrl_quadpid_running_speed(ctrl_t* ctrl, polar_t* command)
 {
     polar_t* speed_order = NULL;
 
-    const pose_t* pose_current = ctrl_get_pose_current(ctrl);
     const polar_t* speed_current = ctrl_get_speed_current(ctrl);
 
     ctrl_quadpid_t* ctrl_quadpid = (ctrl_quadpid_t*)ctrl;
 
     /* get speed order */
     speed_order = ctrl_get_speed_order((ctrl_t*)ctrl_quadpid);
 
-    DEBUG("@robot@,%u,@pose_current@,%.4f,%.4f,%.4f\n",
-                ROBOT_ID,
-                pose_current->x,
-                pose_current->y,
-                pose_current->O);
-
-    DEBUG("@robot@,%u,@speed_current@,%.4f,%.4f\n",
-                ROBOT_ID,
-                speed_current->distance,
-                speed_current->angle);
-
-
```

Comment:
I totally disagree with this move outside of the controller for several reasons:

1) If we want to debug a controller, it is to the controller to provide all the logs required to its debug. I should not need to set ENABLE_DEBUG in platforms and in odometry for this.
2) if I change odometry for IMU for example, I will have to check that they expose logs too to ensure that external tools like simulation are still working.
3) In addition to the several ENABLE_DEBUG I need to activate, log function search or development will be harder if splitted into several files.

What could be done is to add filter keywords with the displayed string:
Instead of "@robot@" we could have "@robot,ctrl@", "robot,ctrl,quadpid@", "@robot,odometry@"

The scope would be clear and if for some reasons logs could be duplicated because of using the same information they could be filtered easily by external tools.
If ENABLE_DEBUG for odometry is not set thus it will not display duplicate information if not needed.

So please lets keep controller stuff into controller scope.

Reply:
Except for displaying numerical values, "pose_current" is not even used in this function which regulates speed only. Code to display this numerical information was factorized.

Anyway, your comments seems related to the way logs are enabled per file in the overall project and the way the controller periodic loop was spread in multiple files, which is out of scope of the proposed work.

>What could be done is to add filter keywords with the displayed string:
>Instead of "@Robot@" we could have "@Robot,ctrl@", "robot,ctrl,quadpid@", "@Robot,odometry@"

This makes me remind a critical point 'totally' dropped during previous refactoring activities : the current implementation is not considering the time overhead introduced by the UART, which is an issue.

Initial implementation was using a RAM based buffer (log_vect) service to first catch numerical value of a cycle between one measurement and one command, and secondly to print those values _after_ command was set, ensuring a minimum time overhead of the numerical value extraction on the controller performance.

https://github.com/cogip/avr-firmware/blob/3e9ed4ebe72fe4e7ad636e01b70284d6e1adf7ec/drivers/system/log.h#L44

In the current design, the overall probes points are adding extra delay which are nearly consuming the 20ms timeframe (around 230 chars printed by uart * 86.4us ~= 19.8 ms), thus adding an extra delay of almost one controller cycle... which is nearly breaking the real-time constraints.

Reply:
Yes indeed, log buffering is mandatory. Lets merge this as is and we could enable log filtering later, putting first the effort on log buffering.